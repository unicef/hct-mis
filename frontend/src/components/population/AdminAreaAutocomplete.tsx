import React, { useEffect, useState } from 'react';
import styled from 'styled-components';
import get from 'lodash/get';
import { InputAdornment } from '@material-ui/core';
import RoomRoundedIcon from '@material-ui/icons/RoomRounded';
import CircularProgress from '@material-ui/core/CircularProgress';
import Autocomplete from '@material-ui/lab/Autocomplete';
import { useDebounce } from '../../hooks/useDebounce';
import TextField from '../../shared/TextField';
import {
  AllAdminAreasQuery,
  useAllAdminAreasLazyQuery,
} from '../../__generated__/graphql';
import { useBusinessArea } from '../../hooks/useBusinessArea';

const StyledAutocomplete = styled(Autocomplete)`
  width: ${(props) => (props.fullWidth ? '100%' : '232px')}
    .MuiFormControl-marginDense {
    margin-top: 4px;
  }
`;

export function AdminAreasAutocomplete({
  disabled,
  fullWidth,
  onFilterChange,
  name,
}: {
  disabled?;
  fullWidth?: boolean;
  onFilterChange;
  name: string;
}): React.ReactElement {
  const [open, setOpen] = useState(false);
  const [inputValue, onInputTextChange] = useState('');

  const debouncedInputText = useDebounce(inputValue, 500);
  const businessArea = useBusinessArea();
  const [loadAdminAreas, { data, loading }] = useAllAdminAreasLazyQuery({
    variables: {
      first: 50,
      title: debouncedInputText,
      businessArea,
      level: 2,
    },
  });
  useEffect(() => {
    loadAdminAreas();
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [debouncedInputText]);

  const onChangeMiddleware = (e, selectedValue): void => {
    onFilterChange((filters) => ({
      ...filters,
      [name]: selectedValue || undefined,
    }));
  };
  return (
    <StyledAutocomplete<AllAdminAreasQuery['allAdminAreas']['edges'][number]>
      fullWidth={fullWidth}
      open={open}
      filterOptions={(options1) => options1}
      onChange={onChangeMiddleware}
      onOpen={() => {
        setOpen(true);
      }}
      onClose={(e, reason) => {
        setOpen(false);
        if (reason === 'select-option') return;
        onInputTextChange('');
      }}
      getOptionSelected={(option, value1) => {
        return value1?.node?.id === option.node.id;
      }}
      getOptionLabel={(option) => {
        if (!option.node) {
          return '';
        }
        return `${option.node.title}`;
      }}
      disabled={disabled}
      options={get(data, 'allAdminAreas.edges', [])}
      loading={loading}
      renderInput={(params) => (
        <TextField
          {...params}
          label='Admin Level 2'
          variant='outlined'
          margin='dense'
          value={inputValue}
          onChange={(e) => onInputTextChange(e.target.value)}
          InputProps={{
            ...params.InputProps,
            startAdornment: (
              <InputAdornment position='start'>
                <RoomRoundedIcon style={{ color: '#5f6368' }} />
              </InputAdornment>
            ),
            endAdornment: (
              <>
                {loading ? (
                  <CircularProgress color='inherit' size={20} />
                ) : null}
                {params.InputProps.endAdornment}
              </>
            ),
          }}
        />
      )}
    />
  );
}
