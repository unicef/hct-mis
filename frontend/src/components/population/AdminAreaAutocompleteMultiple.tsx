import React, { useEffect, useState } from 'react';
import styled from 'styled-components';
import get from 'lodash/get';
import { Box } from '@material-ui/core';
import Autocomplete from '@material-ui/lab/Autocomplete';
import { useDebounce } from '../../hooks/useDebounce';
import TextField from '../../shared/TextField';
import {
  AllAdminAreasQuery,
  useAllAdminAreasQuery,
} from '../../__generated__/graphql';
import { useBusinessArea } from '../../hooks/useBusinessArea';
import { LoadingComponent } from '../LoadingComponent';
import { FieldLabel } from '../FieldLabel';

const StyledAutocomplete = styled(Autocomplete)`
  width: 232px;
  .MuiFormControl-marginDense {
    margin-top: 4px;
  }
`;

export function AdminAreaAutocompleteMultiple({
  value,
  onChange,
  disabled,
}: {
  value;
  onChange;
  disabled?;
}): React.ReactElement {
  const [open, setOpen] = React.useState(false);
  const [inputValue, setInputTextChange] = React.useState('');

  const debouncedInputText = useDebounce(inputValue, 500);
  const [newValue, setNewValue] = useState([]);
  const businessArea = useBusinessArea();
  const { data, loading } = useAllAdminAreasQuery({
    variables: {
      first: 100,
      title: debouncedInputText,
      businessArea,
    },
  });
  useEffect(() => {
    setNewValue(value);
  }, [data, value]);
  useEffect(() => {
    setInputTextChange('');
  }, [value]);

  if (loading) return <LoadingComponent />;
  if (!data) return null;
  return (
    <Box display='flex' flexDirection='column'>
      <FieldLabel>Administrative Level 2</FieldLabel>
      <StyledAutocomplete<AllAdminAreasQuery['allAdminAreas']['edges'][number]>
        open={open}
        multiple
        fullWidth
        filterOptions={(options1) => options1}
        onChange={onChange}
        value={newValue}
        onOpen={() => {
          setOpen(true);
        }}
        onClose={() => {
          setOpen(false);
        }}
        getOptionSelected={(option, value1) => {
          return value1?.node?.id === option.node.id;
        }}
        getOptionLabel={(option) => {
          if (!option.node) {
            return '';
          }
          return `${option.node.title}`;
        }}
        disabled={disabled}
        options={get(data, 'allAdminAreas.edges', [])}
        loading={loading}
        renderInput={(params) => {
          return (
            <TextField
              {...params}
              inputProps={{
                ...params.inputProps,
                value: inputValue,
              }}
              placeholder={
                newValue.length > 0 ? null : 'Administrative Level 2'
              }
              variant='outlined'
              margin='dense'
              value={inputValue}
              onChange={(e) => setInputTextChange(e.target.value)}
            />
          );
        }}
      />
    </Box>
  );
}
