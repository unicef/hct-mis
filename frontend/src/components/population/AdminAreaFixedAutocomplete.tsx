import { InputAdornment } from '@mui/material';
import CircularProgress from '@mui/material/CircularProgress';
import RoomRoundedIcon from '@mui/icons-material/RoomRounded';
import Autocomplete from '@mui/lab/Autocomplete';
import get from 'lodash/get';
import React, { useEffect, useState } from 'react';
import { useTranslation } from 'react-i18next';
import styled from 'styled-components';
import { useBusinessArea } from '../../hooks/useBusinessArea';
import { useDebounce } from '../../hooks/useDebounce';
import TextField from '../../shared/TextField';
import {
  AllAdminAreasQuery,
  useAllAdminAreasQuery,
} from '../../__generated__/graphql';

const StyledAutocomplete = styled(Autocomplete)`
  .MuiFormControl-marginDense {
    //margin-top: 4px;
  }
`;

export function AdminAreaFixedAutocomplete({
  value,
  onChange,
  disabled,
}: {
  value;
  onChange;
  disabled?;
}): React.ReactElement {
  const { t } = useTranslation();
  const [open, setOpen] = React.useState(false);
  const [inputValue, onInputTextChange] = React.useState('');

  const debouncedInputText = useDebounce(inputValue, 500);
  const [newValue, setNewValue] = useState(null);
  const businessArea = useBusinessArea();
  const { data, loading } = useAllAdminAreasQuery({
    variables: {
      name: debouncedInputText,
      businessArea,
      first: 50,
      level: 2,
    },
  });

  useEffect(() => {
    if (data) {
      setNewValue(
        typeof value === 'string'
          ? data.allAdminAreas.edges.find((item) => item.node.name === value)
          : value,
      );
    } else {
      // setNewValue(value);
    }
    // onInputTextChange('');
  }, [data, value]);
  const onChangeMiddleware = (e, selectedValue, reason): void => {
    onInputTextChange(selectedValue?.node?.name);
    onChange(e, selectedValue, reason);
  };
  return (
    <StyledAutocomplete<AllAdminAreasQuery['allAdminAreas']['edges'][number]>
      open={open}
      filterOptions={(options1) => options1}
      onChange={onChangeMiddleware}
      value={newValue}
      onOpen={() => {
        setOpen(true);
      }}
      onClose={(e, reason) => {
        setOpen(false);
        if (value || reason === 'select-option') return;
        onInputTextChange(null);
      }}
      getOptionSelected={(option, selectedValue) => {
        return selectedValue?.node?.id === option.node.id;
      }}
      getOptionLabel={(option) => {
        if (!option.node) {
          return '';
        }
        return `${option.node.name}`;
      }}
      disabled={disabled}
      options={get(data, 'allAdminAreas.edges', [])}
      loading={loading}
      renderInput={(params) => (
        <TextField
          {...params}
          label={t('Administrative Level 2')}
          variant='outlined'
          margin='dense'
          value={inputValue}
          onChange={(e) => onInputTextChange(e.target.value)}
          InputProps={{
            ...params.InputProps,
            startAdornment: (
              <InputAdornment position='start'>
                <RoomRoundedIcon style={{ color: '#5f6368' }} />
              </InputAdornment>
            ),
            endAdornment: (
              <>
                {loading ? (
                  <CircularProgress color='inherit' size={20} />
                ) : null}
                {params.InputProps.endAdornment}
              </>
            ),
          }}
        />
      )}
    />
  );
}
