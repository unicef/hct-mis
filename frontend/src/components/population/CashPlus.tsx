import React from 'react';
import styled from 'styled-components';
import { Typography, Paper, Grid } from '@material-ui/core';
import { LabelizedField } from '../LabelizedField';

const Overview = styled(Paper)`
  padding: ${({ theme }) => theme.spacing(8)}px
    ${({ theme }) => theme.spacing(11)}px;
  margin-bottom: ${({ theme }) => theme.spacing(6)}px;
`;

const Title = styled.div`
  width: 100%;
  padding-bottom: ${({ theme }) => theme.spacing(8)}px;
`;

export function CashPlus({ individual }): React.ReactElement {
  const { enrolledInNutritionProgramme, administrationOfRutf } = individual;
  return (
    <Overview>
      <Title>
        <Typography variant='h6'>Cash+</Typography>
      </Title>
      <Grid container spacing={6}>
        <Grid item xs={4}>
          <LabelizedField
            label='Enrolled in nutrition programme'
            value={enrolledInNutritionProgramme ? 'YES' : 'NO'}
          />
        </Grid>
        <Grid item xs={4}>
          <LabelizedField
            label='Administratiion of rutf'
            value={administrationOfRutf ? 'YES' : 'NO'}
          />
        </Grid>
      </Grid>
    </Overview>
  );
}
