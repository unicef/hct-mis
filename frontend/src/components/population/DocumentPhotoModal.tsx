import React, { useState } from 'react';
import styled from 'styled-components';
import { Button, DialogContent, DialogTitle, Box } from '@material-ui/core';
import { Dialog } from '../../containers/dialogs/Dialog';
import { DialogActions } from '../../containers/dialogs/DialogActions';
import { BlackLink } from '../BlackLink';

const DialogTitleWrapper = styled.div`
  border-bottom: 1px solid ${({ theme }) => theme.hctPalette.lighterGray};
`;

const DialogFooter = styled.div`
  padding: 12px 16px;
  margin: 0;
  border-top: 1px solid ${({ theme }) => theme.hctPalette.lighterGray};
  text-align: right;
`;

const StyledImage = styled.img`
  max-width: 100%;
  max-height: 100%;
`;

interface DocumentPhotoModalProps {
  photo: string;
  documentId: string;
}

export const DocumentPhotoModal = ({
  photo,
  documentId,
}: DocumentPhotoModalProps): React.ReactElement => {
  const [dialogOpen, setDialogOpen] = useState(false);

  return (
    <>
      <BlackLink
        onClick={() => {
          setDialogOpen(true);
        }}
      >
        {documentId}
      </BlackLink>
      <Dialog
        open={dialogOpen}
        onClose={() => setDialogOpen(false)}
        aria-labelledby='form-dialog-title'
      >
        <DialogTitleWrapper>
          <DialogTitle id='scroll-dialog-title'>
            Document&apos;s Photo
          </DialogTitle>
        </DialogTitleWrapper>
        <DialogContent>
          <Box p={3}>
            <StyledImage alt='document' src={photo} />
          </Box>
        </DialogContent>
        <DialogFooter>
          <DialogActions>
            <Button onClick={() => setDialogOpen(false)}>CANCEL</Button>
          </DialogActions>
        </DialogFooter>
      </Dialog>
    </>
  );
};
