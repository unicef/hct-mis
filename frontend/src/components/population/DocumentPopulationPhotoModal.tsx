import { Box, Button, DialogContent } from '@material-ui/core';
import React, { useState } from 'react';
import { useTranslation } from 'react-i18next';
import { Dialog } from '../../containers/dialogs/Dialog';
import { DialogActions } from '../../containers/dialogs/DialogActions';
import {
  IndividualNode,
  useIndividualPhotosLazyQuery,
} from '../../__generated__/graphql';
import { PhotoModalHeader } from '../PhotoModal/PhotoModalHeader';
import {
  DialogFooter,
  StyledImage,
  StyledLink,
} from './IndividualFlexFieldPhotoModal';

interface DocumentPopulationPhotoModalProps {
  individual: IndividualNode;
  documentNumber: string;
  documentId: string;
}

export const DocumentPopulationPhotoModal = ({
  individual,
  documentNumber,
  documentId,
}: DocumentPopulationPhotoModalProps): React.ReactElement => {
  const [turnAngle, setTurnAngle] = useState(90);
  const { t } = useTranslation();
  const [dialogOpen, setDialogOpen] = useState(false);
  const [getPhotos, { data }] = useIndividualPhotosLazyQuery({
    variables: { id: individual?.id },
    fetchPolicy: 'network-only',
  });
  const documentWithPhoto = data?.individual?.documents?.edges?.find(
    (el) => el.node.id === documentId,
  );

  return (
    <>
      <StyledLink
        onClick={() => {
          setDialogOpen(true);
          getPhotos();
        }}
      >
        {documentNumber}
      </StyledLink>
      <Dialog
        open={dialogOpen}
        onClose={() => setDialogOpen(false)}
        aria-labelledby='form-dialog-title'
      >
        <PhotoModalHeader
          turnAngle={turnAngle}
          setTurnAngle={setTurnAngle}
          title="Document's Photo"
        />
        <DialogContent>
          <Box p={3}>
            <StyledImage
              id='modalImg'
              alt='document'
              src={documentWithPhoto?.node?.photo}
            />
          </Box>
        </DialogContent>
        <DialogFooter>
          <DialogActions>
            <Button onClick={() => setDialogOpen(false)}>{t('CANCEL')}</Button>
          </DialogActions>
        </DialogFooter>
      </Dialog>
    </>
  );
};
