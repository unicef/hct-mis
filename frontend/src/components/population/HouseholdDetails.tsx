import React from 'react';
import styled from 'styled-components';
import { Typography, Grid } from '@material-ui/core';
import { LabelizedField } from '../LabelizedField';
import { HouseholdNode } from '../../__generated__/graphql';
import { formatCurrency } from '../../utils/utils';
import { useBusinessArea } from '../../hooks/useBusinessArea';
import { MiśTheme } from '../../theme';
import { ContainerColumnWithBorder } from '../ContainerColumnWithBorder';

const Overview = styled.div`
  display: flex;
  flex-direction: row;
  width: 100%;
`;
const Title = styled.div`
  width: 100%;
  padding-bottom: ${({ theme }) => theme.spacing(8)}px;
`;

const ContentLink = styled.a`
  font-family: ${({ theme }: { theme: MiśTheme }) =>
    theme.hctTypography.fontFamily};
  color: #253b46;
  font-size: 14px;
  line-height: 19px;
`;

interface HouseholdDetailsProps {
  houseHold: HouseholdNode;
}
export function HouseholdDetails({
  houseHold,
}: HouseholdDetailsProps): React.ReactElement {
  const businessArea = useBusinessArea();
  return (
    <ContainerColumnWithBorder>
      <Title>
        <Typography variant='h6'>Details</Typography>
      </Title>
      <Overview>
        <Grid container spacing={6}>
          <Grid item xs={4}>
            <LabelizedField label='Household Size'>
              <div>{houseHold.size}</div>
            </LabelizedField>
          </Grid>
          <Grid item xs={4}>
            <LabelizedField label='Country'>
              <div>{houseHold.country || '-'}</div>
            </LabelizedField>
          </Grid>
          <Grid item xs={4}>
            <LabelizedField label='Residence Status'>
              <div>{houseHold.residenceStatus}</div>
            </LabelizedField>
          </Grid>
          <Grid item xs={4}>
            <LabelizedField label='Country of Origin'>
              <div>{houseHold.countryOrigin}</div>
            </LabelizedField>
          </Grid>
          <Grid item xs={4}>
            <LabelizedField label='Head of Household'>
              <ContentLink
                href={`/${businessArea}/population/individuals/${houseHold.headOfHousehold.id}`}
              >
                {houseHold.headOfHousehold.fullName}
              </ContentLink>
            </LabelizedField>
          </Grid>
          <Grid item xs={4}>
            <LabelizedField label='Total Cash Received'>
              <div>{formatCurrency(houseHold.totalCashReceived)}</div>
            </LabelizedField>
          </Grid>
          <Grid item xs={4}>
            <LabelizedField label='PrOgRAmmE(S) ENROLLED'>
              <div>
                {houseHold.programs.edges.map((item) => (
                  <div>{item.node.name}</div>
                ))}
              </div>
            </LabelizedField>
          </Grid>
          <Grid item xs={4}>
            <LabelizedField label='Administrative Level 1'>
              <div>{houseHold.address || '-'}</div>
            </LabelizedField>
          </Grid>
          <Grid item xs={4}>
            <LabelizedField label='Administrative Level 2'>
              <div>{houseHold.adminArea?.title || '-'}</div>
            </LabelizedField>
          </Grid>
        </Grid>
      </Overview>
    </ContainerColumnWithBorder>
  );
}
