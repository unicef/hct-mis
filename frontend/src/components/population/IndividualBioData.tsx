import React from 'react';
import styled from 'styled-components';
import { Paper, Typography, Grid } from '@material-ui/core';
import { LabelizedField } from '../LabelizedField';
import { IndividualNode } from '../../__generated__/graphql';
import { useHistory } from 'react-router-dom';
import {
  getAgeFromDob,
  sexToCapitalize,
  getIdentificationType, decodeIdString,
} from '../../utils/utils';
import Moment from 'react-moment';
import { useBusinessArea } from '../../hooks/useBusinessArea';

const Overview = styled(Paper)`
  padding: ${({ theme }) => theme.spacing(8)}px
    ${({ theme }) => theme.spacing(11)}px;
`;

const Title = styled.div`
  width: 100%;
  padding-bottom: ${({ theme }) => theme.spacing(8)}px;
`;

const ContentLink = styled.div`
  text-decoration: underline;
  cursor: pointer;
`;

interface IndividualBioDataProps {
  individual: IndividualNode;
}
export function IndividualsBioData({
  individual,
}: IndividualBioDataProps): React.ReactElement {
  const history = useHistory();
  const businessArea = useBusinessArea();

  let age: number | null;
  const { dob } = individual;
  if (dob) {
    age = getAgeFromDob(dob);
  }

  const openHousehold = (): void => {
    history.push(
      `/${businessArea}/population/household/${individual.household.id}`,
    );
  };
  return (
    <Overview>
      <Title>
        <Typography variant='h6'>Bio Data</Typography>
      </Title>
      <Grid container spacing={6}>
        <Grid item xs={4}>
          <LabelizedField label='Full Name'>
            <div>{individual.fullName}</div>
          </LabelizedField>
        </Grid>
        <Grid item xs={4}>
          <LabelizedField label='Given Name'>
            <div>{individual.firstName}</div>
          </LabelizedField>
        </Grid>
        <Grid item xs={4}>
          <LabelizedField label='Middle Name'>
            <div>-</div>
          </LabelizedField>
        </Grid>
        <Grid item xs={4}>
          <LabelizedField label='Family Name'>
            <div>{individual.lastName}</div>
          </LabelizedField>
        </Grid>
        <Grid item xs={4}>
          <LabelizedField label='Sex'>
            <div>{sexToCapitalize(individual.sex)}</div>
          </LabelizedField>
        </Grid>
        <Grid item xs={4}>
          <LabelizedField label='Age'>
            <div>{age}</div>
          </LabelizedField>
        </Grid>
        <Grid item xs={4}>
          <LabelizedField label='Date of Birth'>
            <Moment format='DD/MM/YYYY'>{dob}</Moment>
          </LabelizedField>
        </Grid>
        <Grid item xs={4}>
          <LabelizedField label='Estimated Date of Birth'>
            <div>
              {individual.estimatedDob ? individual.estimatedDob : 'No'}
            </div>
          </LabelizedField>
        </Grid>
        <Grid item xs={4}>
          <LabelizedField label='ID Type'>
            <div>{getIdentificationType(individual.identificationType)}</div>
          </LabelizedField>
        </Grid>
        <Grid item xs={4}>
          <LabelizedField label='ID Number'>
            <div>{individual.identificationNumber}</div>
          </LabelizedField>
        </Grid>
        <Grid item xs={4}>
          <LabelizedField label='Household ID'>
            <ContentLink onClick={() => openHousehold()}>
              {decodeIdString(individual.household.id)}
            </ContentLink>
          </LabelizedField>
        </Grid>
        <Grid item xs={4}>
          <LabelizedField label='Special Privileges'>
            <div>-</div>
          </LabelizedField>
        </Grid>
      </Grid>
    </Overview>
  );
}
