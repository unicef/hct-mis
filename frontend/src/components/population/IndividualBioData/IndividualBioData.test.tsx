import React from 'react';
import { render } from '../../../testUtils/testUtils';
import { fakeHouseholdChoices } from '../../../../fixtures/population/fakeHouseholdChoices';
import { fakeIndividual } from '../../../../fixtures/population/fakeIndividual';
import { IndividualBioData } from './IndividualBioData';

describe('components/population/IndividualBioData', () => {
  it('should render', () => {
    const { container } = render(
      <IndividualBioData
        businessArea='afghanistan'
        individual={fakeIndividual}
        choicesData={fakeHouseholdChoices}
      />,
    );
    expect(container).toMatchSnapshot();
  });
});
