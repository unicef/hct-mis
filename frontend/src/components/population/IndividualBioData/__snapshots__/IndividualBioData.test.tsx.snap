// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/population/IndividualBioData should render 1`] = `
<div>
  <div
    class="MuiPaper-root sc-cSHVUG gnjeDy MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="sc-bxivhb jsIVNl"
    >
      <h6
        class="MuiTypography-root MuiTypography-h6"
      >
        Bio Data
      </h6>
    </div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-6"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <div>
          <span
            class="sc-bwzfXH eoWrid"
            color="textSecondary"
          >
            Full Name
          </span>
          <div
            data-cy="label-Full Name"
          >
            Jan Romaniak
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <div>
          <span
            class="sc-bwzfXH eoWrid"
            color="textSecondary"
          >
            Given Name
          </span>
          <div
            data-cy="label-Given Name"
          >
            Jan
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <div>
          <span
            class="sc-bwzfXH eoWrid"
            color="textSecondary"
          >
            Middle Name
          </span>
          <div
            data-cy="label-Middle Name"
          >
            -
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <div>
          <span
            class="sc-bwzfXH eoWrid"
            color="textSecondary"
          >
            Family Name
          </span>
          <div
            data-cy="label-Family Name"
          >
            Romaniak
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <div>
          <span
            class="sc-bwzfXH eoWrid"
            color="textSecondary"
          >
            Gender
          </span>
          <div
            data-cy="label-Gender"
          >
            Male
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <div>
          <span
            class="sc-bwzfXH eoWrid"
            color="textSecondary"
          >
            Age
          </span>
          <div
            data-cy="label-Age"
          >
            121
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <div>
          <span
            class="sc-bwzfXH eoWrid"
            color="textSecondary"
          >
            Date of Birth
          </span>
          <div
            data-cy="label-Date of Birth"
          >
            <time
              datetime="-2176675200000"
            >
              10 Jan 1901
            </time>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <div>
          <span
            class="sc-bwzfXH eoWrid"
            color="textSecondary"
          >
            Estimated Date of Birth
          </span>
          <div
            data-cy="label-Estimated Date of Birth"
          >
            No
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <div>
          <span
            class="sc-bwzfXH eoWrid"
            color="textSecondary"
          >
            Marital Status
          </span>
          <div
            data-cy="label-Marital Status"
          >
            Married
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <div>
          <span
            class="sc-bwzfXH eoWrid"
            color="textSecondary"
          >
            Work Status
          </span>
          <div
            data-cy="label-Work Status"
          >
            Not provided
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <div>
          <span
            class="sc-bwzfXH eoWrid"
            color="textSecondary"
          >
            Pregnant
          </span>
          <div
            data-cy="label-Pregnant"
          >
            No
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <div>
          <span
            class="sc-bwzfXH eoWrid"
            color="textSecondary"
          >
            Household ID
          </span>
          <div
            data-cy="label-Household ID"
          >
            <a
              class="sc-bdVaJa hqCmJD"
              href="/afghanistan/population/household/SG91c2Vob2xkTm9kZTowMTk1YTEzYi0wYThkLTRiYTEtYjM0OS00MDY4YWNmOTViNDE="
            >
              HH-20-0000.0002
            </a>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <div>
          <span
            class="sc-bwzfXH eoWrid"
            color="textSecondary"
          >
            Role
          </span>
          <div
            data-cy="label-Role"
          >
            None
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <div>
          <span
            class="sc-bwzfXH eoWrid"
            color="textSecondary"
          >
            Relationship to HOH
          </span>
          <div
            data-cy="label-Relationship to HOH"
          >
            Brother / Sister
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <div>
          <span
            class="sc-bwzfXH eoWrid"
            color="textSecondary"
          >
            Preferred language
          </span>
          <div
            data-cy="label-Preferred language"
          >
            -
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <div>
          <span
            class="sc-bwzfXH eoWrid"
            color="textSecondary"
          >
            Linked Households
          </span>
          <div
            data-cy="label-Linked Households"
          >
            -
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="sc-kAzzGY ZtRWH"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <div>
          <span
            class="sc-bwzfXH eoWrid"
            color="textSecondary"
          >
            Observed disabilities
          </span>
          <div
            data-cy="label-Observed disabilities"
          >
            -
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <div>
          <span
            class="sc-bwzfXH eoWrid"
            color="textSecondary"
          >
            Seeing disability severity
          </span>
          <div
            data-cy="label-Seeing disability severity"
          >
            None
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <div>
          <span
            class="sc-bwzfXH eoWrid"
            color="textSecondary"
          >
            Hearing disability severity
          </span>
          <div
            data-cy="label-Hearing disability severity"
          >
            None
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <div>
          <span
            class="sc-bwzfXH eoWrid"
            color="textSecondary"
          >
            Physical disability severity
          </span>
          <div
            data-cy="label-Physical disability severity"
          >
            None
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <div>
          <span
            class="sc-bwzfXH eoWrid"
            color="textSecondary"
          >
            Remembering or concentrating disability severity
          </span>
          <div
            data-cy="label-Remembering or concentrating disability severity"
          >
            None
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <div>
          <span
            class="sc-bwzfXH eoWrid"
            color="textSecondary"
          >
            Self-care disability severity
          </span>
          <div
            data-cy="label-Self-care disability severity"
          >
            None
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <div>
          <span
            class="sc-bwzfXH eoWrid"
            color="textSecondary"
          >
            Communicating disability severity
          </span>
          <div
            data-cy="label-Communicating disability severity"
          >
            None
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <div>
          <span
            class="sc-bwzfXH eoWrid"
            color="textSecondary"
          >
            Disability
          </span>
          <div
            data-cy="label-Disability"
          >
            Not Disabled
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="sc-kAzzGY ZtRWH"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <div
          class="MuiBox-root MuiBox-root-1"
        >
          <div
            class="MuiBox-root MuiBox-root-2"
          >
            <div>
              <span
                class="sc-bwzfXH eoWrid"
                color="textSecondary"
              >
                National ID
              </span>
              <div
                data-cy="label-National ID"
              >
                WSH221365
              </div>
            </div>
          </div>
          <div>
            <span
              class="sc-bwzfXH eoWrid"
              color="textSecondary"
            >
              issued
            </span>
            <div
              data-cy="label-issued"
            >
              Poland
            </div>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="sc-kAzzGY ZtRWH"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <div>
          <span
            class="sc-bwzfXH eoWrid"
            color="textSecondary"
          >
            Email
          </span>
          <div
            data-cy="label-Email"
          >
            -
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <div>
          <span
            class="sc-bwzfXH eoWrid"
            color="textSecondary"
          >
            Phone Number
          </span>
          <div
            data-cy="label-Phone Number"
          >
            +48734123423
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <div>
          <span
            class="sc-bwzfXH eoWrid"
            color="textSecondary"
          >
            Alternative Phone Number
          </span>
          <div
            data-cy="label-Alternative Phone Number"
          >
            -
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12"
      >
        <div
          class="sc-kAzzGY ZtRWH"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-3"
      >
        <div>
          <span
            class="sc-bwzfXH eoWrid"
            color="textSecondary"
          >
            Date of last screening against sanctions list
          </span>
          <div
            data-cy="label-Date of last screening against sanctions list"
          >
            <time
              datetime="1644305739718"
            >
              8 Feb 2022
            </time>
          </div>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-6"
      >
        <div>
          <span
            class="sc-bwzfXH eoWrid"
            color="textSecondary"
          >
            Linked Grievances
          </span>
          <div
            data-cy="label-Linked Grievances"
          >
            <span
              class="sc-htpNat cwQQYq"
              color="textSecondary"
            >
              <span>
                -
              </span>
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
